rfcs/shared/build-system/06-browser-support-for-v9.md

```diff
@@ -0,0 +1,61 @@
+
+### Browser APIs & Expectations
+
+Fluent developers are expected to adhere to the proposed browser matrix thus any usage of browser APIs that aren't compatible should be avoided. For consumers who require the use of older browsers that don't support certain browser features like `ResizeObserver`, they are expected to handle the issue themselves whether through the implementation of their own polyfills or the use of a third-party one.
+
+### CSS Features & Expectations
+
+With a defined browser matrix, the Fluent team will now have better guidance on what CSS features can and cannot be used. Using polyfills is not a real option in this case because the current ecosystem around CSS polyfills results in egregious amounts of code that increase bundle size while also degrading performance. This puts the responsibility squarely on the Fluent team to ensure that no unsupported CSS features are introduced to the library.
+
+### Browser Support Going Forward
+
+With an eye towards being a modern library, Fluent will follow a yearly audit process to evaluate and update the current browser support matrix to keep pace with the ever evolving ecosystem of the web. This will allow the team to utilize modern tools and improve overall engineering efficiency while also providing consumers with performance and bundlesize improvements. Due to the nature of how Safari browser updates are tied to MacOS versions, Safari will be the browser that ultimately holds the library back from fully adopting modern browser versions across the board. Because of this, the proposed timeframe for when these audits occur should be whenever a Safari major release occurs. These tend to happen annually during the month of September so Fluent can target an annual audit around that time.
+
+## Out of Scope
+
+- The specifics on how to prevent the addition of ES syntax, browser APIs or CSS features that don't conform to this matrix.
+
+## Pros and Cons
+
+### Pros
+
+- A source of truth for Fluent team and consumers to refer to for browser support.
+- Able to ship modern ES2019 code with the option of going to ES2020 without downleveling.
+- Less maintanance by putting onus of older/unsupported browsers to consumers.
+
+### Cons
+
+- Incompatible CSS features (`flex gap`) are already present in the library and will need to be removed and refactored.
```

Comment:
I would also add [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) to the list. But we have already used `keyborg` and [its custom selector](https://github.com/microsoft/fluentui/blob/8afe4d15242d6220f202ca1d0d1522f4f9d151bb/packages/react-components/react-tabster/src/focus/createCustomFocusIndicatorStyle.ts#L22).

Reply:
Thanks for pointing this out! Will add that to RFC and from a quick search, we also use this in `react-input`, `react-select`, `react-spinner`, `react-tabster` and `react-textarea`.

Reply:
issue link for this: #23020